contracts/V1/StateChannelDiamondProxy/FraudProofFacet.sol

```diff
@@ -53,11 +50,11 @@ contract FraudProofFacet is StateChannelCommon {
         Block memory block2 = abi.decode(blockDoubleSignProof.block2.encodedBlock, (Block));
 
         if(fraudProofVerificationContext.channelId != block1.transaction.header.channelId || fraudProofVerificationContext.channelId != block2.transaction.header.channelId) {
-            return address(0);
+            revert ErrorNotSameChannelId();
         }
 
-        if(block1.stateSnapshotHash != block2.stateSnapshotHash && block1.previousBlockHash != block2.previousBlockHash) {
-            return address(0);
+        if (block1.transaction.header.forkCnt != block2.transaction.header.forkCnt && block1.transaction.header.transactionCnt != block2.transaction.header.transactionCnt && keccak256(abi.encode(block1)) != keccak256(abi.encode(block2))){
+            revert ErrorDoubleSignBlocksNotSame();
```

Comment:
This logic is not correct - please think about this. If you want to write logic like this I'd advice writing first what is a double sign:
if(fork == fork && height == height && hash != hash) -> this is a double sign. So you either negate this all together as '!(double sign)' and use your 'if - revert' syntax or just use require OR pass through with '!' mathematically:
(!(fork == fork && height == height && hash != hash)) <=> (fork != fork || height != height || hash == hash)

Reply:
Yes the logic follows this
if the fork is same, height is same and the content of the block is same, then its a double sign

Reply:
if the content is the same implicitly the fork and height are the same too and that IS NOT a double sign - that's the same block (blockA and blockA) - a double sign is 2 conflicting blocks (blockA and blockB) by the same author for the same fork and block height.